File: algorithms/2_medium/M/03727_maximum_alternating_sum_of_squares/sort.py
class Solution:
    def maxAlternatingSum(self, nums: List[int]) -> int:
        num_len = len(nums)

        for i in range(num_len):
            nums[i] *= nums[i]

        nums.sort()
        plus_index = num_len//2
        t_sum = 0

        for i in range(plus_index):
            t_sum -= nums[i]

        for i in range(plus_index, num_len):
            t_sum += nums[i]

        return t_sum

File: algorithms/1_easy/C/00217_contains_duplicate/sort.cpp
class Solution {
public:
    static bool containsDuplicate(vector<int>& nums) {
        const ptrdiff_t len = std::ssize(nums);
        sort(nums.begin(), nums.end());

        for (ptrdiff_t i = 1; i < len; i += 1) {
            if (nums[i - 1] == nums[i]) {
                return true;
            }
        }

        return false;
    }
};
